### ISSUE TITLE
Dead connection stuck in pool until timeout

### ISSUE THREAD

**Issue by phemmer** [jackc/pgx#948]
Ran across a situation where dead connections were consuming slots in the pgxpool for a while (15 seconds) before clearing out and allowing new connections.

```go
package main

import (
	"context"
	"fmt"
	"time"

	"github.com/jackc/pgx/v4/pgxpool"
)

func main() {
	ctx := context.Background()
	pool, _ := pgxpool.Connect(ctx, "host=localhost user=postgres pool_max_conns=1")
	pool.Exec(ctx, "SELECT pg_terminate_backend(pg_backend_pid())")
	stat := pool.Stat()
	fmt.Printf("AcquiredConns=%d IdleConns=%d ConstructingConns=%d\n", stat.AcquiredConns(), stat.IdleConns(), stat.ConstructingConns())
	t1 := time.Now()
	pool.Exec(ctx, "SELECT 1")
	t2 := time.Now()
	fmt.Printf("duration: %s\n", t2.Sub(t1).Truncate(time.Millisecond))
}
```
```
AcquiredConns=1 IdleConns=0 ConstructingConns=0
duration: 15.023s
```

**Comment by LeGEC** [comment]
related to https://github.com/jackc/pgconn/issues/65

**Comment by jackc** [comment]
This should be resolved along with  jackc/pgconn#65.
